stream/client.go

```diff
@@ -0,0 +1,85 @@
+package stream
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"net/http"
+
+	"github.com/google/zoekt/rpc"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/query"
+)
+
+type client struct {
+	address string
+	path    string
+	conn    *http.Client
+}
+
+func NewClientAtAddress(address string) *client {
+	rpc.RegisterGob()
+	return &client{address, DefaultSSEPath, &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConns: 500,
+		},
+	}}
+}
+
+type Streamer interface {
+	Send(*zoekt.SearchResult)
+}
+
+// Use StreamerChan to cast a receiving channel of search results to a Streamer.
+type StreamerChan chan<- *zoekt.SearchResult
+
+func (c StreamerChan) Send(result *zoekt.SearchResult) {
+	c <- result
+}
+
+// StreamSearch returns search results as stream via streamer.
+func (c *client) StreamSearch(q query.Q, opts *zoekt.SearchOptions, streamer Streamer) error {
+	// Encode query and opts.
+	buf := new(bytes.Buffer)
+	args := &searchArgs{
+		q, opts,
+	}
+	enc := gob.NewEncoder(buf)
+	err := enc.Encode(args)
+	if err != nil {
+		return fmt.Errorf("error during encoding: %w", err)
+	}
+
+	// Send request.
+	req, err := http.NewRequest("POST", c.address+c.path, buf)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/octet-stream")
```

Comment:
the key for this header should be `Accept` I think?
can use `application/x-your-own-name`